workflow/vm/chain.go

```diff
@@ -0,0 +1,53 @@
+package workflowvm
+
+import (
+	"github.com/mesg-foundation/core/workflow"
+	uuid "github.com/satori/go.uuid"
+)
+
+// executionChain keeps a chain of tasks that should be executed after each other
+// while using previous one's outputs as the next one's inputs.
+type executionChain struct {
+	// w is the workflow of execution chain.
+	w workflow.Workflow
+
+	// index of the current task that has been requested to be executed.
+	index int
+	// indexSecret is a crypto secret that given to current task execution request.
+	indexSecret string
+}
+
+// newChain creates a new execution chain which is not yet in execution process.
+func newChain(w workflow.Workflow) *executionChain {
+	return &executionChain{
+		w:     w,
+		index: -1,
+	}
+}
+
+// IsDone checks whether is the execution chain can create more execution requests or not.
+func (q *executionChain) IsDone() bool {
+	return q.index == len(q.w.Tasks)-1
+}
+
+// Next creates the next execution requests by using Event as its accumulator.
+// Event can be a standard event or a result of previous execution request that in execution chain.
+func (q *executionChain) Next(e Event) (ok bool, exec *Execution) {
```

Comment:
use error (io.EOF) that is used by read/next functions to represent end of data in general.